opty/direct_collocation.py

```diff
@@ -205,13 +218,32 @@ def __init__(self, equations_of_motion, state_symbols,
 
         self.known_parameter_map = known_parameter_map
         self.known_trajectory_map = known_trajectory_map
+        
+        self.num_know_trajectory_map = len(self.known_trajectory_map)
 
         self.instance_constraints = instance_constraints
 
         self.num_constraints = self.num_states * (num_collocation_nodes - 1)
 
         self.tmp_dir = tmp_dir
 
+        self.eom = (1-Lamda_matrix)*equations_of_motion + Lamda_matrix
+		    *(self.state_derivative_symbols - K_matrix*
+		    (known_trajectory_map[self.num_know_trajectory_map 
+		    - self.num_states:]-self.state_symbols))
```

Comment:
If a given state is named `x1`, then we should create new symbols that mirror the states with a name `x_1_m`.